source/s3_client.c

```diff
@@ -180,6 +181,16 @@ struct aws_s3_client *aws_s3_client_new(
         return NULL;
     }
 
+    /* Cannot be less than zero.  If zero, use default. */
+    if (client_config->part_size != 0 && client_config->part_size < MB_TO_BYTES(s_s3_min_part_size_MB)) {
```

Comment:
I tried download with small part_size, it was accpeted by the server. So, the min part size seems only applied to upload? Not sure, we should rule it out totally or not.

Reply:
Yeah, I'm on the fence with that too.